core/src/main/java/org/apache/tomcat/util/net/jss/TomcatJSS.java

```diff
@@ -226,77 +235,83 @@ public void setOcspTimeout(int ocspTimeout) {
         this.ocspTimeout = ocspTimeout;
     }
 
-    public void loadJSSConfig(String jssConf) throws Exception {
+    public void loadJSSConfig(String jssConf) throws IOException {
         File configFile = new File(jssConf);
         loadJSSConfig(configFile);
     }
 
-    public void loadJSSConfig(File configFile) throws Exception {
+    public void loadJSSConfig(File configFile) throws IOException {
 
         Properties config = new Properties();
-        config.load(new FileReader(configFile));
+        try (FileReader fr = new FileReader(configFile)) {
+            config.load(fr);
+        }
 
         loadJSSConfig(config);
     }
 
-    public void loadJSSConfig(Properties config) throws Exception {
+    public void loadJSSConfig(Properties config) {
 
-        String certDb = config.getProperty("certdbDir");
-        if (certDb != null)
-            setCertdbDir(certDb);
+        String certDbProp = config.getProperty("certdbDir");
+        if (certDbProp != null)
+            setCertdbDir(certDbProp);
 
```

Comment:
Probably not. I tried to avoid making functional changes to the code, but I could probably tidy this up.

I have made some functional change to the code though, clearly, as the CI now fails!

Reply:
The error is server generated by a server time-out. It did not  start although I cannot spot changes which could generate the problem!

Reply:
The only place where I can think I functionally changed anything was the renaming of the shadowed fields. It is possible that something is wrongly configured here. I have restored the original names, so we will see whether that is indeed the case shortly.

Reply:
It is not, it is something else. I will have to go through and roll back changes to see what I have broken.

Reply:
I figured it out! The redundant `return` that was removed was not so redundant.

The `iteration` variable is only updated in the `catch` block, so if the `return` statement is removed you sit in an infinite loop of successful logins.

To get around this I introduced a Boolean to carry the state of the login success and use that in the loop condition.